src/worker.js

```diff
@@ -284,6 +287,7 @@ function handleMessage(e) {
       }
     } else if (e.data.cmd === 'cancel') { // Main thread is asking for a pthread_cancel() on this thread.
       if (Module['_pthread_self']()) {
+        closeProxyBrokerPort();
```

Comment:
I think these need to go through `Module`, like the calls around them. `worker.js` is a separate file from the main JS.

Or, we can just add a proxy closing function here perhaps if it is short and trivial?